src/System.Text.Json/src/System/Text/Json/Node/JsonNumber.cs

```diff
@@ -7,89 +7,357 @@
+        public JsonNumber(sbyte value)
+        {
+            SetSByte(value);
+        }
         [CLSCompliant(false)]
-        public JsonNumber(ushort value) { }
+        public JsonNumber(ushort value)
+        {
+            SetUInt16(value);
+        }
         [CLSCompliant(false)]
-        public JsonNumber(uint value) { }
+        public JsonNumber(uint value)
+        {
+            SetUInt32(value);
+        }
         [CLSCompliant(false)]
-        public JsonNumber(ulong value) { }
+        public JsonNumber(ulong value)
+        {
+            SetUInt64(value);
+        }
 
-        public string GetString() { throw null; }
-        public byte GetByte() { throw null; }
-        public int GetInt32() { throw null; }
-        public short GetInt16() { throw null; }
-        public long GetInt64() { throw null; }
-        public float GetSingle() { throw null; }
-        public double GetDouble() { throw null; }
+        public string GetString()
+        { 
+            return _value; 
+        }
+        public byte GetByte()
+        {
+            return byte.Parse(_value);
+        }
+        public short GetInt16()
+        {
+            return short.Parse(_value);
+        }
+        public int GetInt32()
+        {
+            return int.Parse(_value);
+        }
+        public long GetInt64()
+        {
+            return long.Parse(_value);
+        }
+        public float GetSingle()
+        {
+            return float.Parse(_value);
+        }
+        public double GetDouble()
+        {
+            return double.Parse(_value);
+        }
         [CLSCompliant(false)]
-        public sbyte GetSByte() { throw null; }
+        public sbyte GetSByte()
+        {
+            return sbyte.Parse(_value);
+        }
         [CLSCompliant(false)]
-        public ushort GetUInt16() { throw null; }
+        public ushort GetUInt16()
+        {
+            return ushort.Parse(_value);
+        }
         [CLSCompliant(false)]
-        public uint GetUInt32() { throw null; }
+        public uint GetUInt32()
+        {
+            return uint.Parse(_value);
+        }
         [CLSCompliant(false)]
-        public ulong GetUInt64() { throw null; }
+        public ulong GetUInt64()
+        {
+            return ulong.Parse(_value);
+        }
 
-        public bool TryGetByte(out byte value) { throw null; }
-        public bool TryGetInt32(out int value) { throw null; }
-        public bool TryGetInt16(out short value) { throw null; }
-        public bool TryGetInt64(out long value) { throw null; }
-        public bool TryGetSingle(out float value) { throw null; }
-        public bool TryGetDouble(out double value) { throw null; }
+        public bool TryGetByte(out byte value)
+        {
+            try
+            {
+                value = GetByte();
+                return true;
+            } 
+           catch
+            {
+                value = 0;
+                return false;
```

Comment:
Try methods shouldn't be written as catch/ignore/return-false.  Implementing a non-Try method in terms of the Try is legitimate, though (the question I thought you asked me in person earlier).

But in this case, as @joperezr pointed out, the ideal implementation is to just defer to byte.TryParse (et al).

Reply:
As a general rule, try not to use exceptions (try/catch) as control flow:
https://softwareengineering.stackexchange.com/questions/189222/are-exceptions-as-control-flow-considered-a-serious-antipattern-if-so-why